satpy/tests/test_dataset.py

```diff
@@ -250,7 +250,8 @@ def test_combine_real_world_mda(self):
                                                 'cpp_reff_pal',
                                                 '-'],
                         'platform_name': 'NOAA-20',
-                        'sensor': {'viirs'}},
+                        'sensor': {'viirs'},
+                        'raw_metadata': {'foo': np.array([1, 2, 3])}},
```

Comment:
It is possible to make these test values more complicated to mimic the original issue? I'm guessing as-is, this would pass before or after this PR.

Reply:
A dictionary with numpy arrays already triggers the original issue.